apps/dav/lib/Command/ExportCalendar.php

```diff
@@ -0,0 +1,95 @@
+<?php
+
+declare(strict_types=1);
+/**
+ * SPDX-FileCopyrightText: 2025 Nextcloud GmbH and Nextcloud contributors
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+namespace OCA\DAV\Command;
+
+use InvalidArgumentException;
+use OCA\DAV\CalDAV\Export\ExportService;
+use OCP\Calendar\CalendarExportOptions;
+use OCP\Calendar\ICalendarExport;
+use OCP\Calendar\IManager;
+use OCP\IUserManager;
+use Symfony\Component\Console\Attribute\AsCommand;
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputArgument;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+
+/**
+ * Calendar Export Command
+ *
+ * Used to export data from supported calendars to disk or stdout
+ */
+#[AsCommand(
+	name: 'calendar:export',
+	description: 'Export calendar data from supported calendars to disk or stdout',
+	hidden: false
+)]
+class ExportCalendar extends Command {
+	public function __construct(
+		private IUserManager $userManager,
+		private IManager $calendarManager,
+		private ExportService $exportService,
+	) {
+		parent::__construct();
+	}
+
+	protected function configure(): void {
+		$this->setName('calendar:export')
+			->setDescription('Export calendar data from supported calendars to disk or stdout')
+			->addArgument('uid', InputArgument::REQUIRED, 'Id of system user')
+			->addArgument('cid', InputArgument::REQUIRED, 'Id of calendar')
+			->addArgument('format', InputArgument::OPTIONAL, 'Format of output (iCal, jCal, xCal) default to iCal')
```

Comment:
```suggestion
			->addArgument('format', InputArgument::OPTIONAL, 'Format of output (iCal, jCal, xCal) default to iCal', 'ical')
```

For addArgument with InputArgument::OPTIONAL it's possible  to define a default. You can use that to simplfiy the code below from

```
		if ($format !== null && !in_array($format, ExportService::FORMATS, true)) {
			throw new InvalidArgumentException("Format <$format> is not valid.");
		} else {
			$options->setFormat($format ?? 'ical');
		}
```

to


```
		if (!in_array($format, ExportService::FORMATS, true)) {
			throw new InvalidArgumentException("Format <$format> is not valid.");
		}
		$options->setFormat($format);
```

Reply:
NIce! I wasn't aware defaults on command where a thing! thank you